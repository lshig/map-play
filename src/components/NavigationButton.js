import PropType from 'prop-types';
import React from 'react';

export default function NavigationButton({ id, onClick, label }) {
  return (
    <h3 className="mapNavigation" id={id} onClick={onClick}>
      {label}
    </h3>
  );
}

NavigationButton.propTypes = {
  id: PropType.string.isRequired,
  label: PropType.string.isRequired,
  onClick: PropType.func.isRequired
};
