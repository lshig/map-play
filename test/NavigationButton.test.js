import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavigationButton from '../src/components/NavigationButton';

describe('NavigationButton', () => {
  describe('basic', () => {
    const wrapper = shallow(
      <NavigationButton id="foobar" label="foobar-button" onClick={jest.fn()} />
    );

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
