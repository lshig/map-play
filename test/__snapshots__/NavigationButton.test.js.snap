// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButton basic renders 1`] = `
<h3
  className="mapNavigation"
  id="foobar"
  onClick={[MockFunction]}
>
  foobar-button
</h3>
`;
